sdk/forms/src/common/component.ts

```diff
@@ -1,10 +1,14 @@
-export function component<P extends object, E extends Node>(
+export function component<P extends object, E extends Node & { unmount?: () => void }>(
   renderer: (props: P) => E,
   didChange = defaultDidChangeFunction<P>,
 ) {
   let prev: { props: P; element: E };
   return (props: P) => {
     if (prev && !didChange(prev.props, props)) return prev.element;
+    if (prev && didChange(prev.props, props)) {
+      // before re-rendering, unmount the previous element
+      prev.element?.unmount?.();
+    }
```

Comment:
Not sure exactly how to do it but this solution is not recursively calling unmount for nested elements.

For example with the `DisplayValue` component if the value is an object then it adds `DisplayValue` children but when the root `DisplayValue` is unmounted it doesn't call unmount for nested children.

Reply:
After some discussion we've decided to not support event handlers for objects. Only primitives are supported for now. So currently this is not a concern.